### PR TITLE
Use :1337 if no port is specified when using "localhost"

### PR DESCRIPTION
**Pull request by nunopato**
- when using "localhost" without specifying a port, default to :1337

### CODE REVIEW

**Review by plmercereau — CHANGES_REQUESTED** [review]
Please add a changeset

**Comment by plmercereau** [comment]
subdomain: 'localhost:1234' is somehow semantically incorrect. We can live with it, but I think we should introduce a `port` param at a later stage, that would default to `1337` when `subdomain` equals `'localhost'`

**Comment by elitan** [comment]
Update potential docs that currently use `localhost:1337` to `localhost`. Either after merging or in this PR.

https://github.com/nhost/nhost/pull/783#pullrequestreview-1027664862

**Comment by elitan** [comment]
> subdomain: 'localhost:1234' is somehow semantically incorrect. We can live with it, but I think we should introduce a `port` param at a later stage, that would default to `1337` when `subdomain` equals `'localhost'`

This should work because there are developers who are using `subdomain: 'localhost:1337'` with the current version and things should not break for them just because of this update.

File: packages/nhost-js/src/utils/helpers.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@ export function urlFromParams(
   }
 
   if (backendOrSubdomain.subdomain !== undefined && backendOrSubdomain.subdomain.match(LOCALHOST)) {
-    return `http://${backendOrSubdomain.subdomain}/v1/${service}`
+    return backendOrSubdomain === 'localhost'
+      ? `http://localhost:1337/v1/${service}`
+      : `http://${backendOrSubdomain.subdomain}/v1/${service}`
```
Comment (elitan) [inline packages/nhost-js/src/utils/helpers.ts:24]: For developers who are currently using `localhost:1337` this update will break their code.

I think we should allow for `localhost:<port>` here. Something like this:


```js
return backendOrSubdomain === 'localhost'
      ? `http://localhost:1337/v1/${service}`
      : backendOrSubdomain.startsWith('localhost:') (or regex) 
        ? `http://${backendOrSubdomain}/v1/${service}` 
        : `http://${backendOrSubdomain.subdomain}/v1/${service}`
```

**Review by elitan — CHANGES_REQUESTED** [review]

**Comment by plmercereau** [comment]
`examples/react-apollo/cypress/videos/1-unauthenticated/authentication-guards.cy.ts.mp4` etc should not be pushed

**Comment by elitan** [comment]
> `examples/react-apollo/cypress/videos/1-unauthenticated/authentication-guards.cy.ts.mp4` etc should not be pushed

Should be fixed here: https://github.com/nhost/nhost/pull/786/commits/9feea987e68a3be00c8bc860d1e16fff7193bc59

File: packages/nhost-js/src/utils/helpers.ts [hunk 2]
```diff
@@ -14,17 +14,38 @@ export function urlFromParams(
...
-  if (backendOrSubdomain.subdomain !== undefined && backendOrSubdomain.subdomain.match(LOCALHOST)) {
-    return `http://${backendOrSubdomain.subdomain}/v1/${service}`
+  // to make TS happy
+  if (!subdomain) {
+    throw new Error('`subdomain` must be set if `backendUrl` is not set.')
+  }
+
+  // check if subdomain is localhost[:port]
+  const subdomainLocalhostFound = subdomain.match(LOCALHOST_REGEX)
+  if (subdomainLocalhostFound && subdomainLocalhostFound.length > 0) {
```
Comment (plmercereau) [inline packages/nhost-js/src/utils/helpers.ts:34]: could be simplified by
```js
if (LOCALHOST_REGEX.test(subdomain)
```

**Review by plmercereau — APPROVED** [review]

**Review by elitan — APPROVED** [review]

**Pull request merged**